src/networking/APNSNetworkService.php

```diff
@@ -0,0 +1,119 @@
+<?php
+declare( strict_types = 1 );
+
+class APNSNetworkService {
+
+	/** @var resource **/
+	private $curl_handle;
+
+	/** @var int */
+	public $port = 443;
+
+	/** @var bool **/
+	private $debug = false;
+
+	/** @var bool **/
+	private $ssl_verification_enabled = true;
+
+	public function __construct() {
+		$ch = curl_multi_init();
+		curl_multi_setopt( $ch, CURLMOPT_PIPELINING, CURLPIPE_MULTIPLEX );
+		curl_multi_setopt( $ch, CURLMOPT_MAX_TOTAL_CONNECTIONS, 1 );
+		curl_multi_setopt( $ch, CURLOPT_PIPEWAIT, true );
+
+		$this->curl_handle = $ch;
+	}
+
+	public function setPort( int $port ): self {
```

Comment:
I used the same chainable setter style that `APNSClient` uses, even though we might not need it here, to be consistent.

I can remove it if you think the syntax overhead is not worth it.

Reply:
IMHO it doesn't hurt, and this way it's consistent with the rest of the library

Reply:
Perfectomundo.